### PR TITLE
Code change for accepting project name and library strategy from command line for collapse-rnaseq

### PR DESCRIPTION
**Pull request by sangeetashukla**
### Purpose:
Update collapse-rnaseq script to accept values for project name and library strategy

### Directions for reviewers:
Please review the code change and verify if the script runs without error

### CODE REVIEW

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh [hunk 1]
```diff
@@ -19,17 +19,54 @@ cd "$script_directory" || exit
...
 for strategy in ${libraryStrategies[@]}; do
 
   Rscript --vanilla 01-summarize_matrices.R \
-    -i ../../data/pbta-gene-expression-rsem-fpkm.${strategy}.rds \
+    -i ../../data/${project_acronym}-gene-expression-rsem-fpkm.${strategy}.rds \
     -g ../../data/gencode.v27.primary_assembly.annotation.gtf.gz \
-    -m results/pbta-gene-expression-rsem-fpkm-collapsed.${strategy}.rds \
-    -t results/pbta-gene-expression-rsem-fpkm-collapsed_table.${strategy}.rds
+    -m results/${project_acronym}-gene-expression-rsem-fpkm-collapsed.${strategy}.rds \
+    -t results/k${project_acronym}fnbl-gene-expression-rsem-fpkm-collapsed_table.${strategy}.rds
```
Comment (kgaonkar6) [inline analyses/collapse-rnaseq/run-collapse-rnaseq.sh]:
```suggestion
    -t results/${project_acronym}-gene-expression-rsem-fpkm-collapsed_table.${strategy}.rds
```

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh [hunk 2]
```diff
@@ -19,17 +19,54 @@ cd "$script_directory" || exit
...
 
 done
 
 # run the notebook for analysis of dropped genes
-Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.polya.rds', stranded.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.stranded.rds'), clean = TRUE)"
+#Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/${project_acronym}-gene-expression-rsem-fpkm-collapsed_table.polya.rds', stranded.annot.table = 'results/${project_acronym}-gene-expression-rsem-fpkm-collapsed_table.stranded.rds'), clean = TRUE)"
```
Comment (kgaonkar6) [inline analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: this should be uncommented, and you can add a output file name as well:

```suggestion
#Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/${project_acronym}-gene-expression-rsem-fpkm-collapsed_table.polya.rds', stranded.annot.table = 'results/${project_acronym}-gene-expression-rsem-fpkm-collapsed_table.stranded.rds'), output_file = paste0('02-analyze-drops-', '${project_acronym}', clean = TRUE)"
```

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh [hunk 3]
```diff
@@ -19,17 +19,54 @@ cd "$script_directory" || exit
...
+
+if [ -z "${libraryStrategies}" ]; then
+    usage
+fi
+
+
 # generate collapsed matrices for poly-A and stranded datasets
-libraryStrategies=("polya" "stranded")
+#libraryStrategies=("polya" "stranded")
+#libraryStrategies=("stranded")
```
Comment (kgaonkar6) [inline analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: I think you can now remove these commented code:
```suggestion

```

**Review by kgaonkar6 — CHANGES_REQUESTED** [review]
Thanks for updating the bash script in the module, looks good.  The code runs successfully, however 1 output file is named as ` kkfnblfnbl-gene-expression-rsem-fpkm-collapsed_table.stranded.rds` which seems unwanted. I've added some suggestion to solve that issue.  

Also, I think the 02-analyze-drops.Rmd should be run but seems to be commented out right now can we uncomment that and rerun?

**Comment by jharenza** [comment]
Hi @sangeetashukla - will you also run this for the tpm matrix and add the collapsed matrix here? That way, @komalsrathi can use it tomorrow to generate a figure for the Wednesday meeting. Thank you!

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh (hunk 1 above)
Comment (sangeetashukla) [reply analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: @kgaonkar6 Thanks for finding that. I have fixed that line.

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh (hunk 2 above)
Comment (sangeetashukla) [reply analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: @kgaonkar6 Sure, I have now uncommented the markdown call.

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh (hunk 3 above)
Comment (sangeetashukla) [reply analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: @kgaonkar6 Sounds good. I have removed the commented lines.

**Comment by sangeetashukla** [comment]
Hi @jharenza @komalsrathi @kgaonkar6 I have updated the bash script to accept TPM file as input, and have run this for KFNBL TPM. 


The resulting collapsed TPM matrices are also in this branch.
[https://github.com/sangeetashukla/OpenPBTA-analysis/blob/update-collapse-rnaseq-1-1/analyses/collapse-rnaseq/results/kfnbl-gene-expression-rsem-tpm-collapsed.stranded.rds](url)
[https://github.com/sangeetashukla/OpenPBTA-analysis/blob/update-collapse-rnaseq-1-1/analyses/collapse-rnaseq/results/kfnbl-gene-expression-rsem-tpm-collapsed_table.stranded.rds](url)

Please take a look and let me know in case there are any issues.

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh [hunk 4]
```diff
@@ -19,17 +19,58 @@ cd "$script_directory" || exit
...
-    -m results/pbta-gene-expression-rsem-fpkm-collapsed.${strategy}.rds \
-    -t results/pbta-gene-expression-rsem-fpkm-collapsed_table.${strategy}.rds
+    -m results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed.${strategy}.rds \
+    -t results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.${strategy}.rds\
 
 done
 
 # run the notebook for analysis of dropped genes
-Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.polya.rds', stranded.annot.table = 'results/pbta-gene-expression-rsem-fpkm-collapsed_table.stranded.rds'), clean = TRUE)"
+Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', params = list(polya.annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.polya.rds', stranded.annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.stranded.rds'), clean = TRUE)"
```
Comment (kgaonkar6) [inline analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: Thanks for the updated. However, we don't have a `polya.annot.table` so the notebook errors out. So I suggest we might have to update the 02-analyze-drops.Rmd (to make it more flexible to to able to use only stranded data as we have in kfnbl ) to run per `strategy` like:

```
for strategy in ${libraryStrategies[@]}; do

  Rscript --vanilla 01-summarize_matrices.R \
    -i ../../data/${project_acronym}-gene-expression-rsem-${quantificationType}.${strategy}.rds \
    -g ../../data/gencode.v27.primary_assembly.annotation.gtf.gz \
    -m results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed.${strategy}.rds \
    -t results/k${project_acronym}fnbl-gene-expression-rsem-${quantificationType}-collapsed_table.${strategy}.rds

    # run the notebook for analysis of dropped genes
    Rscript -e "rmarkdown::render(input = '02-analyze-drops.Rmd', output_file = paste0('02-analyze-drops-', '${project_acronym}','-${quantificationType}'),params = list(annot.table = 'results/${project_acronym}-gene-expression-rsem-${quantificationType}-collapsed_table.${strategy}.rds' , strategy='${strategy}'), clean = TRUE)"
done

```

Comment (kgaonkar6) [reply analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: I could not make a pull request to your forked repo, so I've saved the updated 02-analyze-drops.Rmd code here https://gist.github.com/kgaonkar6/b0f982e59857f258b6f10b0aadd03105 to run per strategy and quantification type.

**Comment by kgaonkar6** [comment]
Also a heads up, in v2 data release `BS_ETC8R0TD` will be added to rnaseq files so this module will need a rerun with v2 files.

**Comment by jharenza** [comment]
@sangeetashukla can you download v2 data and rerun this?

**Comment by komalsrathi** [comment]
@sangeetashukla we would also need expected counts collapsed to gene symbol. cc: @jharenza

File: analyses/collapse-rnaseq/run-collapse-rnaseq.sh (hunk 4 above)
Comment (sangeetashukla) [reply analyses/collapse-rnaseq/run-collapse-rnaseq.sh]: Thanks, @kgaonkar6 . I have updated the script.

**Comment by sangeetashukla** [comment]
> @sangeetashukla can you download v2 data and rerun this?

@jharenza @kgaonkar6 
Sure. I downloaded the v2 files, and re-ran the script. 
Updated scripts and result files are committed to my branch and filed with this PR.

**Comment by sangeetashukla** [comment]
> @sangeetashukla we would also need expected counts collapsed to gene symbol. cc: @jharenza

@komalsrathi I have updated the script to also accept expected_count as input file and process into gene symbol. The branch filed in this PR has the updated scripts and the result files for kfnbl expected counts.

Please review and let me know in case there are any issues.

**Comment by komalsrathi** [comment]
> > @sangeetashukla we would also need expected counts collapsed to gene symbol. cc: @jharenza
> 
> @komalsrathi I have updated the script to also accept expected_count as input file and process into gene symbol. The branch filed in this PR has the updated scripts and the result files for kfnbl expected counts.
> 
> Please review and let me know in case there are any issues.

Thank you! I'll check and let you know.

**Review by kgaonkar6 — APPROVED** [review]
Looks perfect to me, thank you @sangeetashukla!
Could you please update the README in this module? 

Also it seems your branch is not up-to-date with the `dev` branch in this repo.
Have you set up `PediatricOpenTargets/OpenPBTA-analysis` as a remote upstream repository? You can follow these instructions for configuring a remote that points to an upstream repository here if you have not done this already:
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/configuring-a-remote-for-a-fork

Then we would basically have to get latest changes from upstream and merge to your forked repo. This is psuedo git commands which I think will help:

```
# fetch the changes from the PediatricOpenTargets/OpenPBTA-analysis repository
git fetch upstream

# checkout your own master branch
git checkout master

# merge the upstream/master branch into your own master branch
git merge upstream/master

# push the changes to your master branch to the remote repository
git push

# Since we want to use the `dev` branch as default in your forked repo as well
git checkout dev 
git merge master
git push

# update your update-collapse-rnaseq-1-1
git checkout update-collapse-rnaseq-1-1
git merge dev
git push
```

**Comment by sangeetashukla** [comment]
> Looks perfect to me, thank you @sangeetashukla!
> Could you please update the README in this module?
> 
> Also it seems your branch is not up-to-date with the `dev` branch in this repo.
> Have you set up `PediatricOpenTargets/OpenPBTA-analysis` as a remote upstream repository? You can follow these instructions for configuring a remote that points to an upstream repository here if you have not done this already:
> https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/configuring-a-remote-for-a-fork
> 
> Then we would basically have to get latest changes from upstream and merge to your forked repo. This is psuedo git commands which I think will help:
> 
> ```
> # fetch the changes from the PediatricOpenTargets/OpenPBTA-analysis repository
> git fetch upstream
> 
> # checkout your own master branch
> git checkout master
> 
> # merge the upstream/master branch into your own master branch
> git merge upstream/master
> 
> # push the changes to your master branch to the remote repository
> git push
> 
> # Since we want to use the `dev` branch as default in your forked repo as well
> git checkout dev 
> git merge master
> 
> # update your update-collapse-rnaseq-1-1
> git checkout update-collapse-rnaseq-1-1
> git merge dev
> ```

@kgaonkar6 
Great. I will update the README.
I have set up `PediatricOpenTargets/OpenPBTA-analysis` as upstream repository. I will go ahead and merge the latest changes to my forked repo.

**Comment by kgaonkar6** [comment]
Thanks for fixing the conflict and updating the README. Looks good.

**Pull request merged**